frontend/components/book-table-page/BooktablePage.tsx

```diff
@@ -94,20 +96,32 @@ export function BooktablePage({ restaurant }: BooktablePageProps) {
   const { timeLeft, isRunning, startTimer } = useTimer(60); 
   
   const onSubmit = async (data: FormData) => {
-    if (verifyOtp) {
-        try {
-            await sanityClient.create({
-                _type: 'reservation',
-                ...data, 
-            });
-        } catch (error) {
-            console.error('Error creating reservation:', error);
-            alert('Failed to book a table. Please try again later.');
-        }
-    } else {
-        alert('Please verify your phone number');
+    const result = await checkAvailability(tables, data.guests, data.date, data.time, verifyOtp);
+    if (result.status === 'error') {
+      alert(result.message);
+      return;
     }
-  };
+  
+    const reservationData = {
+      ...data,
+      table: {
+        _type: 'reference',
+        _ref: result.table, 
+      },
+      time: `${data.date}T${data.time}`,
+    };
+  
+    try {
+      await sanityClient.create({
+        _type: 'reservation',
+        ...reservationData,
+      });
+      alert('Table booked successfully!');
```

Comment:
Don't use alert. Create a popup or grab one from the ui library. The default alters look bad